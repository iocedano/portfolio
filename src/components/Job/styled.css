.job-content {
  font-family: Open Sans;
  /* text-align: center;
  padding: 10px;
  opacity: 0;
  transition: all .5s ease-in-out;
  transform: translateY(20px);
  cursor: pointer; */
}

.job-company {
  font-size: 14px;
  color: #fff;
}

.job-title {
  font-size: 14px;
  color: #fff;
}
