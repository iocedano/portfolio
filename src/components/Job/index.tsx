import React, { memo } from "react";
import PropTypes, { InferProps } from 'prop-types';
import './styled.css';

export const JobPropTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  id: PropTypes.string,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  company: PropTypes.string.isRequired
}

export type JobTypes = InferProps<typeof JobPropTypes>;

const Job = ({
  title,
  description,
  company
}: JobTypes) => {
  const randomColor =  Math.floor(Math.random()*16777215).toString(16);

  return(
    <div className="job-content" data-id-test="a-job-test" style={{ backgroundColor: `#${randomColor}`}}>
      <h3 className="job-company">{company}</h3>
      <h4 className="job-title">Role: {title}</h4>
      {description && (
        <p className="job-description">{description}</p>
      )}
    </div>
  );
};

Job.propTypes = JobPropTypes;

Job.defaultProps = {
  description: '',
  id: ''
};

export default memo(Job);
