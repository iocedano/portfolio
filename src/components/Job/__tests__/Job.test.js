import React from 'react';
import { render } from '@testing-library/react';
import SingleJobFixture from '../__fixtures__/single.json';
import Job from '..';

describe('Job', () => {
  it('should render a single job', () => {
    const { title, description, company } = SingleJobFixture;
    const { container } = render(<Job title={title} description={description} company={company} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render description if not present', () => {
    const { container } = render(<Job title="title"  company="test" />);

    expect(container.querySelector('p')).toBeNull();
  })
});
