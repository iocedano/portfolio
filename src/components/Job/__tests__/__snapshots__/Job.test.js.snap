// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job should render a single job 1`] = `
<div
  class="job-content"
  data-id-test="a-job-test"
>
  <h3
    class="job-company"
  >
    test
  </h3>
  <h4
    class="job-title"
  >
    Role: 
    test
  </h4>
  <p
    class="job-description"
  >
    test
  </p>
</div>
`;
